docs/upgrade.md

```diff
@@ -88,6 +88,15 @@ process, for example:
 
 # Upgrading to v1.39.0
 
+## Deprecation of `additional_resources`
+
+The `additional_resources` section of the listener configuration is deprecated in favour
+of the new modules system introduced in Synapse v1.37.0. Authors of modules using this
```

Comment:
It may be worth mentioning that modules can migrate away from `additional_resources` without needing to use the new modules system? For instance, if I were making use of `additional_resources` for a `PresenceRouter` module, I shouldn't think that I need to convert my module first (which currently isn't possible).

Reply:
> It may be worth mentioning that modules can migrate away from `additional_resources` without needing to use the new modules system?

They do need to use `register_web_resource` which is part of the new system though. Though yeah might be worth rewording so maintainers don't think they need to port their entire module first.